test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/MockTracingTelemetry.java

```diff
@@ -40,7 +44,19 @@ public TracingContextPropagator getContextPropagator() {
 
     @Override
     public void close() {
-        ((StrictCheckSpanProcessor) spanProcessor).ensureAllSpansAreClosed();
-        ((StrictCheckSpanProcessor) spanProcessor).clear();
-    }
+        //Sleeping for 2s to get all the spans emitted.
+        try {
+            Thread.sleep(2000);
```

Comment:
Would this result in making all ITs wait for 2s?

Reply:
Added this to wait for all spans to get emitted. But yes this is not needed here as we have mocked telemetry plugin. Added this in OTel specific integ test (Where the delay is 2s).